doc/api/embedding.md

```diff
@@ -167,8 +167,1327 @@ int RunNodeInstance(MultiIsolatePlatform* platform,
 }
 ```
 
+# C embedder API
+
+<!--introduced_in=REPLACEME-->
+
+While Node.js provides an extensive C++ embedding API that can be used from C++
+applications, the C-based API is useful when Node.js is embedded as a shared
+libnode library into C++ or non-C++ applications.
+
+The C embedding API is defined in [src/node_embedding_api.h][] in the Node.js
+source tree.
+
+## API design overview
+
+One of the goals for the C based embedder API is to be ABI stable. It means that
+applications must be able to use newer libnode versions without recompilation.
+The following design principles are targeting to achieve that goal.
```

Comment:
This is good, but for the first version I think we need to add some text to clarify that while the goal is that we end up with an ABI stable API, that is not part of what is being promised yet.